packages/rule-http-compression/src/rule.ts

```diff
@@ -456,6 +492,16 @@ export default class HttpCompressionRule implements IRule {
                      * not being served with `content-encoding: gzip`.
                      */
 
+                    const safeRawResponse = asyncTry<Buffer>(response.body.rawResponse);
```

Comment:
Actually I was thinking about changing this at the connector level

Reply:
@molant Can you provide more details?

Reply:
The connector should have the try/catch and return null if it fails so we don't have to wrap it in here.

Reply:
@molant Can you open an issue?